pkg/controller/utils/datadog/metrics_forwarder.go

```diff
@@ -184,13 +184,17 @@ func (mf *metricsForwarder) stop() {
 	close(mf.stopChan)
 }
 
-func (mf *metricsForwarder) getStatus() *v1alpha1.DatadogAgentCondition {
+// ConditionInterface is a generic interface to encompass DatadogAgentCondition (v1alpha1) and metav1.Condition (v2alpha1)
+type ConditionInterface interface {
```

Comment:
I like the idea, but do we only look at conditions in the status? Should we name it `StatusInterface`

Reply:
we talked separately, since it only deals with condition here going to keep the name as is. (in the next PR maybe i'll rename the function to `getCondition`)

Reply:
make sense

Reply:
do you plan to define the interface in a following PR?

Reply:
as far as i know there's nothing to define, since the condition doesn't have any actions?